_commands/tools/grep.md

```diff
@@ -3,22 +3,100 @@
+
+~~~ bash
+$ grep -v "^[aeiou]" animals.txt
+// finds and prints all animals that do not begin with a lower-case vowel
+$ grep -c -v "s" animals.txt        // counts all animals that don't contain an "s"
+17460
+~~~
+
+#### -i, -&#45;ignore-case
+
+##### Regular expressions, by their nature, are case sensitive. Sometimes, it can be annoying to construct a regex that inherently ignores case differences between letters; this can be especially true when working with extended ASCII or Unicode characters. The `-i` or `--ignore-case` option handles this automatically.
+
+~~~ bash
+$ grep -i "^A" animals.txt
+// finds and prints all animals that begin with an "a" or an "A"
+~~~
+
+#### Other Useful Options
+
+##### `-n`, `--line-number` will list the line number of the matching line before the contents
+`-l`, `--files-with-matches` will list the input files that contain a matching line, not the matching lines themselves
+
+### Input/Output
+
+##### `grep` can be used with one file or with multiple files by listing more than one in the input position for the command. You can even use regular expressions to match multiple files!
+
+~~~ bash
+$ grep "umich" file1.txt file2.txt       // search two files explicitly
+// output here
+$ grep "umich" *.txt                 // search all .txt files in the current directory
+// output here
+$ grep "umich" .                         // search all files in the current directory
+~~~
 
-##### Break it down
+Input can also be piped into `grep` from another program or redirected using &lt;. Likewise, the output of `grep` can be piped into another program (including another `grep`) or redirected using &gt;
 
-#### Example command
+### Regular Expression Basics
 
-##### Break it down
+##### Here are some useful symbols and techniques for regular expressions. This is just a very basic overview; regular expressions are extremely powerful if you know how to wield them.
+* ^ matches the beginning of a line
+    * "^A" will match all lines beginning with a capital "A"
+* $ matches the end of a line
+    * "e$" will match all lines ending with a lower-case "e"
+* . matches any single character (including whitespace)
+    * "b.b" will match all lines that contain two lower-case "b"s separated by a single character (i.e. "bob" or "b&b")
+* &#42; will match the preceding character zero or more times
+    * "a*b" will match all lines that contain any number of lower-case "a"s (including 0) followed by a lower-case "b"
+    * "blue.&#42;green" will match all lines that contain "blue" followed at some point later by "green"
+* &#43; will match the preceding character one or more times
+    * "a&#43;b" will match all lines that contain at least one (but possibly more) lower-case "a" immediately followed by a lower-case "b"
+    * "blue.&#43;green" will match all lines that contain "blue" followed later by "green" with at least one character in between
+* /w will match any letter, digit, or the underscore
```

Comment:
Should these be forward (/) or backslashes (\)? I thought they should be backslashes, but I could be wrong.

Reply:
Yeah, that was my bad: hit the wrong slash button. This should be fixed now.